src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,261 @@
+using Microsoft.Azure.WebJobs.Host.Storage.Blob;
+using Microsoft.Azure.WebJobs.Host.Timers;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.Azure.WebJobs.Host.Blobs.Listeners
+{
+    internal sealed class ScanBlobScanLogHybridPollingStrategy : IBlobListenerStrategy
+    {
+        private static readonly TimeSpan _pollintInterval = TimeSpan.FromSeconds(10);
+        private enum PollStrategy { ScanBlobs, ScanLogs, NotDetermined };
+        private int _scanBlobLimitPerPoll = 10000;
+        private readonly IDictionary<IStorageBlobContainer, ICollection<ITriggerExecutor<IStorageBlob>>> _registrations;
+        private readonly IDictionary<IStorageBlobContainer, DateTime> _lastSweepCycleStartTime;
+        private readonly IDictionary<IStorageBlobContainer, DateTime> _currentSweepCycleStartTime;
+        private readonly ConcurrentQueue<IStorageBlob> _blobsFoundFromScanOrNotification;
+        private readonly IDictionary<IStorageBlobContainer, BlobContinuationToken> _continuationTokens;
+        private PollLogsStrategy _pollLogStrategy;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private bool _disposed;
+
+        public ScanBlobScanLogHybridPollingStrategy() : base()
+        {
+            _registrations = new Dictionary<IStorageBlobContainer, ICollection<ITriggerExecutor<IStorageBlob>>>(
+                new StorageBlobContainerComparer());
+            _pollLogStrategy = new PollLogsStrategy();
+            _cancellationTokenSource = new CancellationTokenSource();
+            _continuationTokens = new Dictionary<IStorageBlobContainer, BlobContinuationToken>();
+            _lastSweepCycleStartTime = new Dictionary<IStorageBlobContainer, DateTime>(
+                new StorageBlobContainerComparer());
+            _currentSweepCycleStartTime = new Dictionary<IStorageBlobContainer, DateTime>(
+                new StorageBlobContainerComparer());
+            _blobsFoundFromScanOrNotification = new ConcurrentQueue<IStorageBlob>();
+        }
+
+        public void Start()
+        {
+            ThrowIfDisposed();
+            _pollLogStrategy.Start();
+        }
+
+        public void Cancel()
+        {
+            ThrowIfDisposed();
+            _pollLogStrategy.Cancel();
+            _cancellationTokenSource.Cancel();
+        }
+
+        public async Task RegisterAsync(IStorageBlobContainer container, ITriggerExecutor<IStorageBlob> triggerExecutor, CancellationToken cancellationToken)
+        {
+            // Register and Execute are not concurrency-safe.
+            // Avoiding calling Register while Execute is running is the caller's responsibility.
+            ThrowIfDisposed();
+
+            // Register all in logPolling, there is no problem if we get 2 notifications of the new blob
+            Task registerLogPolling = _pollLogStrategy.RegisterAsync(container, triggerExecutor, cancellationToken);
```

Comment:
from a simplicity perspective, can we just do the await here rather than run it in parallel with the rest of this function?  This is parallelize with in-memory dictionary lookups, which seems low value.